### PR TITLE
refactor(0.3.0): use dojo 0.3.0 syntax

### PR DESCRIPTION
**Pull request by credence0x**
some notes
- `CreateLaborAuction` was removed in favor of `config_systems.set_labor_auction`
-  `DefineHyperStructure` was removed in favor of `config_systems.create_hyperstructure`
-  `order` folder was renamed to `trade` and has the following interfaces
```
trait ITradeSystems<TContractState> {
    fn create_order(...) -> ID;
    fn accept_order(...);
    fn claim_order(...);
    fn cancel_order(...);
}

trait ITradeCaravanSystems<TContractState> {
    fn attach_caravan(...);
}
```
- `transport_units, caravan, road and travel systems` can now be found in the `transport` folder.

also, many unused systems / code was deleted. we can always find them in source control

### CODE REVIEW

File: contracts/src/systems/transport/contracts/caravan_systems.cairo [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            // DISCUSS: i could also create a new CheckAllSamePosition system that checks
+            // if a list of entities are at the same position, but again that would be 
+            // an extra loop
+            let average_speed = total_speed / total_quantity;
+            let average_speed: u16 = average_speed.try_into().unwrap();
+
+            // set the caravan entity
+            set!(world, (
+                    Owner {
+                        entity_id: caravan_id.into(),
```
Comment (ponderingdemocritus) [inline contracts/src/systems/transport/contracts/caravan_systems.cairo:106]: the `caravan_id.into()` should be cached above for all the following model updates - saves gas

File: contracts/src/systems/transport/contracts/road_systems.cairo [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    impl RoadSystemsImpl of IRoadSystems<ContractState> {
+
+        fn create(
+            self:@ContractState, world: IWorldDispatcher, 
+            entity_id: u128, start_coord: Coord, end_coord: Coord, usage_count: usize
+        ) {
+
+            // assert that entity is owned by caller
+            let entity_owner = get!(world, entity_id, Owner);
+            assert(entity_owner.address == starknet::get_caller_address(), 'entity id not owned by caller');
```
Comment (ponderingdemocritus) [inline contracts/src/systems/transport/contracts/road_systems.cairo:26]: For the future - we could turn this function into a new Cairo Component and share across contracts

File: contracts/src/systems/transport/contracts/transport_unit_systems.cairo [hunk 3]
```diff
@@ -0,0 +1,110 @@
+#[system]
```
Comment (ponderingdemocritus) [inline contracts/src/systems/transport/contracts/transport_unit_systems.cairo:1]: Add comments to this

Comment (ponderingdemocritus) [reply contracts/src/systems/transport/contracts/transport_unit_systems.cairo:1]: explaining the system

File: contracts/src/systems/transport/contracts/travel_systems.cairo [hunk 4]
```diff
@@ -0,0 +1,76 @@
+#[system]
```
Comment (ponderingdemocritus) [inline contracts/src/systems/transport/contracts/travel_systems.cairo:1]: comments

File: contracts/src/systems/transport/contracts/transport_unit_systems.cairo (hunk 3 above)
Comment (credence0x) [reply contracts/src/systems/transport/contracts/transport_unit_systems.cairo:1]: added

**Pull request merged**